src/main.rs

```diff
@@ -30,34 +38,59 @@ impl Terminal {
         }
     }
 
-    fn should_ask_for_todo(&mut self) -> bool {
+    fn should_ask_for_todo(&mut self) -> Result<bool, TerminalError> {
         let mut buf = String::new();
-        self.stdin.read_line(&mut buf).unwrap();
 
-        buf.trim() == "s"
+        self.stdin.read_line(&mut buf).map_err(TerminalError::StdoutErr)?;
+
+        Ok(buf.trim() == "s")
     }
 
-    fn ask_for_new_todo(&mut self) -> Todo {          
+    fn ask_for_new_todo(&mut self) -> Result<Todo, TerminalError> {
 
         println!("\nVocê gostaria de adicionar um novo TODO? 🤔 (Digite: 's' para SIM ou qualquer outra tecla para NÃO)");
-        
-        let response = self.should_ask_for_todo();
-        
-        if !response {
+                
+        if let Ok(false) = self.should_ask_for_todo() {
```

Comment:
Hmm, e o possível erro retornado aqui?
```suggestion
        if !self.should_ask_for_todo()? {
```

Reply:
Show